src/scripts/containers/UploadFileContainer.js

```diff
@@ -1,9 +1,50 @@
 import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+
+import React, { Component } from 'react'
+
+import type { RootState } from 'types/ApplicationTypes'
+
+import { upload } from 'actions/UploadActions'
 
 import UploadFile from 'components/UploadFile'
 
-const mapStateToProps = () => ({})
+type Props = {
+  upload: (filename: string) => void
+}
+
+class UploadFileContainer extends Component<Props, void> {
+  constructor (props) {
+    super(props)
+    this.state = {file: null}
+    this.handleFileChosen = this.handleFileChosen.bind(this)
+    this.handleUploadRequested = this.handleUploadRequested.bind(this)
+  }
+
+  handleFileChosen (file) {
+    this.setState({
+      file: file
+    })
+  }
+
+  handleUploadRequested () {
+    this.props.upload(this.state.file)
+  }
+
+  render () {
+    return (
+      <UploadFile
+        onFileChosen={this.handleFileChosen}
+        onUploadRequested={this.handleUploadRequested}
+      />
+    )
+  }
+}
+
+const mapStateToProps = (state: RootState) => ({})
 
-const mapDispatchToProps = () => ({})
+const mapDispatchToProps = dispatch => ({
+  upload: bindActionCreators(upload, dispatch)
+})
 
-export default connect(mapStateToProps, mapDispatchToProps)(UploadFile)
+export default connect(mapStateToProps, mapDispatchToProps)(UploadFileContainer)
```

Comment:
This isn't a huge point, but thus far we have been separating the `containers` from the `components`. In effect, this would mean moving the component class to a `components/UploadFile.js` module and importing it into this container. Then this container would just define the `mapState/DispatchToProps` and `connect` call.

One of the benefits of doing this is so that it is easier to unit test the components (but that's not really a huge deal). A more important point is that we should just decide which way to do it, because as it stands we are using two different techniques in the same repo.